pkg/query/endpointset.go

```diff
@@ -44,12 +46,25 @@ type queryConnMetricLabel string
 const (
 	ExternalLabels queryConnMetricLabel = "external_labels"
 	StoreType      queryConnMetricLabel = "store_type"
+	GroupKey       queryConnMetricLabel = "group_key"
+	ReplicaKey     queryConnMetricLabel = "replica_key"
 )
 
+var gReplicaKeySuffixRegex *regexp.Regexp
+
+func init() {
+	gReplicaKeySuffixRegex = regexp.MustCompile(`-rep[0-9]$`)
```

Comment:
aha, this seem very specific to our setup :)

Reply:
yes, it's a hacky solution. 
<img width="891" alt="image" src="https://github.com/databricks/thanos/assets/129140017/b4bab5b7-7e5c-4b4e-876a-def8ca9797e0">